pkg/proxy/util/nodeport_addresses_test.go

```diff
@@ -31,7 +32,24 @@ type InterfaceAddrsPair struct {
 	addrs []net.Addr
 }
 
-func TestGetNodeAddresses(t *testing.T) {
+func checkNodeIPs(expected sets.Set[string], actual []net.IP) error {
```

Comment:
sets.Set already has method for Difference(), Equal(), Intersection(), are those not able to achieve the same?

Reply:
We have to do a loop over `actual` to convert the IPs to strings, so at that point it seemed easier to just construct `notFound` and `extra` by hand...